templates/terraform/encoders/bigquery_dataset.go.erb

```diff
@@ -0,0 +1,9 @@
+// `datasetId`. and `project` need to be nested within `datasetReference`
+config := meta.(*Config)
+
+dataRefObj := make(map[string]interface{})
+dataRefObj["datasetId"] = d.Get("dataset_id").(string)
+dataRefObj["projectId"], _ = getProject(d, config)
```

Comment:
This should return the error instead of swallowing it

Reply:
File is now gone since we use `flattenObject`.